src/App.tsx

```diff
@@ -15,48 +16,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  Name = 'name',
+  Lenght = 'length',
+  Default = '',
+}
+
+interface SortingOptions {
+  sortType: SortType,
+  isReversed: boolean,
+}
+
+function getPreparedGoods(
+  goods: string[],
+  {
+    sortType,
+    isReversed,
+  }: SortingOptions,
+): string[] {
+  let preparedGoods = [...goods];
+
+  if (sortType) {
+    switch (sortType) {
+      case SortType.Name:
+        preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+
+      case SortType.Lenght:
+        preparedGoods
+          .sort((good1, good2) => good1[sortType] - good2[sortType]);
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  if (isReversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortType, setSortType] = useState(SortType.Default);
+  const [isReversed, setReverseField] = useState(false);
```

Comment:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```

We need to name it next style -> `[name, setName] = useState();`